usl_pipeline/cloud_functions/main.py

```diff
@@ -903,3 +903,104 @@ def _write_as_npy(blob: storage.Blob, array: NDArray) -> None:
     npy_file.flush()
     npy_file.seek(0)
     blob.upload_from_file(npy_file)
+
+
+def _start_feature_rescaling_if_ready(feature_bucket: storage.Bucket, blob_path: str):
+    """Creates scaled version for every chunk file in the storage bucket parent folder.
+
+    Output files with scaled feature matrices will be placed to the same folder where
+    unscaled files (following "chunk_*" name pattern) are stored but will get "scaled_"
+    prefix. Rescaling only starts when metadata of study area (corresponding to the
+    parent folder name) contains chunks counters giving exactly the same number of
+    chunks as are currently stored in the bucket folder. This should happen when the
+    last feature chunk is stored.
+
+    Args:
+        feature_bucket: Storage bucket with feature matrices.
+        blob_path: Path pointing to one of unscaled feature matrix files with the name
+            following the "chunk_*" pattern. Other files are ignored.
+    """
+    study_area_name, chunk_name = _parse_chunk_path(blob_path)
+    # Let's look up chunk metadata and check if scaling is needed
+    db = firestore.Client()
+    chunk_metadata = metastore.StudyAreaSpatialChunk.get_if_exists(
+        db, study_area_name, chunk_name
+    )
+    if chunk_metadata is None:
+        logging.info(
+            f"Chunk metadata is not registered for {study_area_name}/{chunk_name}"
+        )
+        return
+
+    if not chunk_metadata.needs_scaling:
+        logging.info(f"Chunk {study_area_name}/{chunk_name} doesn't need scaling")
+        return
+    study_area = metastore.StudyArea.get(db, study_area_name)
+
+    if blob_path.endswith(".npy"):
+        if study_area.chunk_x_count is None or study_area.chunk_y_count is None:
+            raise ValueError(
+                f"[Feature Rescaler] Study area {study_area_name} has no chunk counts"
+                + " in metadata"
+            )
+
+        feature_blobs = list(
+            feature_bucket.list_blobs(prefix=f"{study_area_name}/chunk_")
+        )
+
+        chunk_count = study_area.chunk_x_count * study_area.chunk_y_count
+        if chunk_count != len(feature_blobs):
+            raise ValueError(
```

Comment:
Is this an error condition, or are we just waiting for the rest of the files to be written?

Reply:
Oh, sorry, you're right. It was originally done as logging and then I lost track a bit. Added a comment here to keep the context later.